src/components/LabelMapper/DeliverablesAnalysisProgressStatusLabelMapper.tsx

```diff
@@ -0,0 +1,30 @@
+import { DeliverableAnalyzerOperation } from 'pnc-api-types-ts';
+
+import { ILabelMapper, LabelMapper } from 'components/LabelMapper/LabelMapper';
+
+const DELIVERABLES_ANALYSIS_PROGRESS_STATUSES: ILabelMapper<DeliverableAnalyzerOperation['progressStatus']> = {
+  NEW: {
+    text: 'NEW',
+    color: 'grey',
+  },
+  PENDING: {
+    text: 'PENDING',
+    color: 'grey',
+  },
+  IN_PROGRESS: {
+    text: 'IN PROGRESS',
+    color: 'blue',
+  },
+  FINISHED: {
+    text: 'FINISHED',
+    color: 'green',
+  },
+};
+
+interface IDeliverablesAnalysisStatusLabelProps {
+  progressStatus: Exclude<DeliverableAnalyzerOperation['progressStatus'], undefined>;
```

Comment:
I can see sometimes `undefined` is excluded and sometimes it's not excluded. When it's needed?

Reply:
This was originally written by Patrik here: https://github.com/project-ncl/pnc-web-ui-react/blame/3e0ff31fa4e5c389741e3a31944f70c8e1e303c4/src/components/DeliverablesAnalysisProgressStatusLabel/DeliverablesAnalysisProgressStatusLabel.tsx#L26.
@patrikk0123 would you like to provide some explain or it was just a mistake?

Reply:
@matedo1 I had a quick check, The reason for excluding undefined from `DeliverablesAnalysisProgressStatusLabelMapper` and `DeliverablesAnalysisResultLabelMapper` is that these two fields are optional from the backend, while other mappers are mandatory thus no need to worry about undefined values.

Reply:
So WDYT is the best approach now?

Reply:
I think it's fine for our scenarios.

Reply:
done